src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts

```diff
@@ -640,17 +640,25 @@ export class AttachContextAction extends Action2 {
 		const defer = new DeferredPromise<boolean>();
 		const addPromises: Promise<void>[] = [];
 
-		store.add(qp.onDidAccept(e => {
+		store.add(qp.onDidAccept(async e => {
 			const [selected] = qp.selectedItems;
 			if (isChatContextPickerPickItem(selected)) {
 				const attachment = selected.asAttachment();
+				if (!attachment || attachment === 'noop') {
+					return;
+				}
 				if (isThenable(attachment)) {
-					addPromises.push(attachment.then(v => widget.attachmentModel.addContext(v)));
+					addPromises.push(attachment.then(v => widget.attachmentModel.addContext(v as IChatRequestVariableEntry)));
```

Comment:
In the interface we say that you can return `Promise<IChatRequestVariableEntry | 'noop'>`. We should either change the interface to not allow that (making the `as` cast unnecessary) or support it here